pkg/ipamd/datastore/data_store.go

```diff
@@ -832,32 +836,53 @@ func (ds *DataStore) unassignPodIPAddressUnsafe(addr *AddressInfo) {
 	assignedIPs.Set(float64(ds.assigned))
 }
 
-// GetStats returns total number of IP addresses, number of assigned IP addresses, total prefixes and IPs in cooldown period
-func (ds *DataStore) GetStats(addressFamily string) (int, int, int, int) {
+type DataStoreStats struct {
+	// Total number of addresses allocated
+	TotalIPs int
+	// Total number of prefixes allocated
+	TotalPrefixes int
+
+	// Number of assigned addresses
+	AssignedIPs int
+	// Number of addresses in cooldown
+	CooldownIPs int
+}
+
+func (stats DataStoreStats) String() string {
+	return fmt.Sprintf("Total IPs/Prefixes = %d/%d, AssignedIPs/CooldownIPs: %d/%d",
+		stats.TotalIPs, stats.TotalPrefixes, stats.AssignedIPs, stats.CooldownIPs)
+}
+
+func (stats *DataStoreStats) AvailableAddresses() int {
+	return stats.TotalIPs - stats.AssignedIPs
+}
+
+// GetStats returns DataStoreStats for addressFamily
+func (ds *DataStore) GetStats(addressFamily string) DataStoreStats {
```

Comment:
[Non-blocking] nit: Can we call this GetIPStats ? or change GetIPStatsFromCidr to GetStatsFromCidr so we have consistent meaning about Stats.

Reply:
renamed to GetIPStats